### PR TITLE
DOC-571: table header type changes

### PR DESCRIPTION
**Pull request by tylerkelly13**
Related Ticket: DOC-571

Description of Changes:
* Placeholder text

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: _data/nav.yml [hunk 1]
```diff
@@ -479,6 +479,7 @@
     - url: "#API"
     - url: "#Events"
     - url: "#Commands"
+    - url: "#Queryable values"
```
Comment (lnewson) [inline _data/nav.yml:482]: "Query commands" or "Query value commands" might make more sense? "Queryable values" sounds rather wrong/incorrect IMO and we should have "command" in there somewhere.

File: _includes/commands/advtable-cmds.md [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 ```js
 tinymce.activeEditor.execCommand('mceAdvancedTableSort')
-tinymce.activeEditor.execCommand('mceSortTableAdvanced', { sortby: 'row' ; roworcol: '2' ; sort: 'table' ; order: 'ascending' ; })
+tinymce.activeEditor.execCommand('mceSortTableAdvanced', false, { sortby: 'row', roworcol: '2', sort: 'table', order: 'ascending' })
```
Comment (lnewson) [inline _includes/commands/advtable-cmds.md:22]: Oops, how did we miss that last time?

File: _data/nav.yml (hunk 1 above)
Comment (lnewson) [reply _data/nav.yml:482]: I see later you used "Queryable command values" which would be better, but I still find "Queryable" odd 🤷

File: advanced/editor-command-identifiers.md [hunk 3]
```diff
@@ -402,4 +424,92 @@ tinymce.execCommand('mceToggleEditor', false, '<editor_id>');
...
+
+### Listing core and plugin queryable command states
+
+To retrieve a list of avaliable queryable states from the active editor, run the following command from the browser console:
+
+```js
+tinymce.activeEditor.editorCommands.commands.state;
+```
+
+### Available queryable command states
```
Comment (lnewson) [inline advanced/editor-command-identifiers.md:441]: The case of many of these commands are incorrect (technically doesn't matter though as the lookup is case insensitive). Anyways, it should be the same as the `execCommand` name, which for most of these are title case with the exception of ones starting with "mce", as they are camel case.

You can see the correct case in the source as well if that helps: https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/api/EditorCommands.ts#L542

File: advanced/editor-command-identifiers.md [hunk 4]
```diff
@@ -402,4 +424,92 @@ tinymce.execCommand('mceToggleEditor', false, '<editor_id>');
...
+
+### Listing core and plugin queryable command values
+
+To retrieve a list of avaliable queryable command values from the active editor, run the following command from the browser console:
+
+```js
+tinymce.activeEditor.editorCommands.commands.value;
+```
+
+### Available queryable command values
```
Comment (lnewson) [inline advanced/editor-command-identifiers.md:495]: Likewise the case for the command name here is incorrect.

File: plugins/table.md [hunk 5]
```diff
@@ -361,7 +361,7 @@ tinymce.init({
 
 ### `table_style_by_css`
 
-This option enables you to force Table Properties dialog to use HTML5/CSS3 standards for setting cell spacing and cellpadding. By default, these are added as attributes to the table element. By setting this to true, cell spacing is applied to the table element as a `border-spacing` style and cell padding is applied to all `td` elements as a `padding` style
+This option enables you to force Table Properties dialog to use HTML5/CSS3 standards for setting cell spacing and cellpadding. By default, these are added as attributes to the table element. By setting this to true, cell spacing is applied to the table element as a `border-spacing` style and cell padding is applied to all `td` elements as a `padding` style.
```
Comment (lnewson) [inline plugins/table.md:364]: "force the Table Properties"?

File: plugins/table.md [hunk 6]
```diff
@@ -380,6 +380,94 @@ tinymce.init({
   table_style_by_css: false
 });
 ```
+
+### `table_header_type`
```
Comment (lnewson) [inline plugins/table.md:384]: Nice work on this! 👍

File: _includes/commands/advtable-cmds.md (hunk 2 above)
Comment (tylerkelly13) [reply _includes/commands/advtable-cmds.md:22]: apparently :(

File: plugins/table.md (hunk 5 above)
Comment (tylerkelly13) [reply plugins/table.md]: Fixed by https://github.com/tinymce/tinymce-docs/pull/1590/commits/c4f6e844ae7168e5343fd3e281d776e5355b09ac

File: advanced/editor-command-identifiers.md (hunk 4 above)
Comment (tylerkelly13) [reply advanced/editor-command-identifiers.md]: fixed by https://github.com/tinymce/tinymce-docs/pull/1590/commits/c4f6e844ae7168e5343fd3e281d776e5355b09ac

File: advanced/editor-command-identifiers.md (hunk 3 above)
Comment (tylerkelly13) [reply advanced/editor-command-identifiers.md]: Fixed by https://github.com/tinymce/tinymce-docs/pull/1590/commits/c4f6e844ae7168e5343fd3e281d776e5355b09ac

File: _data/nav.yml (hunk 1 above)
Comment (tylerkelly13) [reply _data/nav.yml]: Fixed by https://github.com/tinymce/tinymce-docs/pull/1590/commits/c4f6e844ae7168e5343fd3e281d776e5355b09ac

File: _data/nav.yml [hunk 7]
```diff
@@ -666,10 +667,9 @@
   - url: "php-upload-handler"
   - url: "editor-command-identifiers"
     pages:
-      - url: "#Listing core and plugin editor commands"
-      - url: "#Core Editor commands"
-      - url: "#Plugin Commands"
-      - url: "#Editor Management Commands"
+      - url: "#Executable commands"
+      - url: "#Query command states"
+      - url: "#Query value commands"
```
Comment (lnewson) [inline _data/nav.yml:672]: `Query command values`? As this and state really should be consistent.

File: release-notes/release-notes54.md [hunk 8]
```diff
@@ -30,6 +30,8 @@ The {{site.productname}} 5.4 release includes the following improvements for the
 - Adds: commands, APIs, and icons for; cut, copy, and paste columns.
 - Adds toolbar button icons for the cut, copy, and paste rows.
 - Adds a new `mceTableApplyCellStyle`command for applying selected styles to table cells.
+- Adds a new `table_header_type` option for setting the structure used for table header rows.
```
Comment (metricjs) [inline release-notes/release-notes54.md:33]: "HTML structure" to be slightly more specific?

File: plugins/table.md [hunk 9]
```diff
@@ -380,6 +380,94 @@ tinymce.init({
...
+
+The `table_header_type` option affects how tables are structured when a table row is set as a header row. Note that this setting does not affect header columns.
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply plugins/table.md:386]: no, not sure we will support it.

File: release-notes/release-notes54.md (hunk 8 above)
Comment (tylerkelly13) [reply release-notes/release-notes54.md]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1590/commits/966fedd38b68c8897d01d21eefeb077e14ea954c

File: plugins/table.md [hunk 10]
```diff
@@ -380,6 +380,94 @@ tinymce.init({
...
+      </tr>
+    </tbody>
+  </table>
+  ```
+
+- `auto` - Finds the first existing header row in the table and uses the same structure. If no other table header row exists, the `section` header type will be applied.
+
+**Type:** `String`
+
+**Default Value:** `'auto'`
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply plugins/table.md]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1590/commits/966fedd38b68c8897d01d21eefeb077e14ea954c

File: _includes/commands/table-cmds.md [hunk 11]
```diff
@@ -26,6 +26,9 @@
 | mceTableProps           | Opens the Table Properties dialog.              |
 | mceTableRowProps        | Opens the table row properties dialog.          |
 | mceTableCellProps       | Opens the table cell properties dialog.         |
+| mceTableRowType       | Changes the current row or rows to the specified type, either: `'header'`, `'body'`, or `'footer'`. {{site.requires_5_4v}} |
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply _includes/commands/table-cmds.md]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1590/commits/966fedd38b68c8897d01d21eefeb077e14ea954c

File: _includes/commands/table-cmds.md [hunk 12]
```diff
@@ -26,6 +26,9 @@
...
+| mceTableColType       | Changes the current column or columns to the specified type, either: `'td'` or `'th'`. {{site.requires_5_4v}} |
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply _includes/commands/table-cmds.md]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1590/commits/966fedd38b68c8897d01d21eefeb077e14ea954c

File: _data/nav.yml (hunk 7 above)
Comment (tylerkelly13) [reply _data/nav.yml]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1590/commits/966fedd38b68c8897d01d21eefeb077e14ea954c

**Pull request merged**